docs/en/operations/system-tables/query_log.md

```diff
@@ -61,6 +61,7 @@ Columns:
 - `databases` ([Array](../../sql-reference/data-types/array.md)([LowCardinality(String)](../../sql-reference/data-types/lowcardinality.md))) — Names of the databases present in the query.
 - `tables` ([Array](../../sql-reference/data-types/array.md)([LowCardinality(String)](../../sql-reference/data-types/lowcardinality.md))) — Names of the tables present in the query.
 - `columns` ([Array](../../sql-reference/data-types/array.md)([LowCardinality(String)](../../sql-reference/data-types/lowcardinality.md))) — Names of the columns present in the query.
+- `columns_in_where` ([Array](../../sql-reference/data-types/array.md)([LowCardinality(String)](../../sql-reference/data-types/lowcardinality.md))) — Names of the columns present in the query where conditions.
```

Comment:
To be honest, I don't like the new field. It seems not systematic. Someone else may add `columns_in_select`, `columns_in_order_by`, `columns_in_group_by` etc., and then do the same for `databases`, `tables`, `projections` etc. In the end, the entire query_log will blow up.

Reply:
In many scenarios involving complex tables and BI-generated complex queries, users are often unsure about setting appropriate primary keys and ORDER BY keys. We aim to automatically adjust the ORDER BY clause based on user queries to optimize query efficiency. Therefore, columns_in_where are extremely useful for us. Alternatively, could you provide suggestions on how to implement this? @rschu1ze

Reply:
The use case you describe is about recommending a primary key for a table based on the workload (SELECT queries).

This is a well-known problem and the best solution is to observe (e.g. keep statistics about) filter operations during query execution. The problem with using the raw query strings to suggest a primary key is that the query optimizer may (in principle) rewrite/optimize the query in arbitrary ways such that the filter columns in the SELECT query (WHERE clause) no longer corresponds to the actually performed filter operations. Unfortunately, ClickHouse does not have such a feature right now, it would be cool to implement that of course.

Reply:
YES, We're currently working on it and hope to add 'columns_in_where' to avoid a large amount of redundant and incorrect offline query analysis.

Reply:
unfortunately, this CI has been a bit problematic, and I'm still working on it. Could you please continue with the review...

Reply:
I am personally not a big fan of extending `query_log`, it will be a quick and dirty solution for a problem that needs to be solved more comprehensively. But that is just my opinion. Other reviewers may have a different view.

Reply:
Yes, I'm thinking of another way, planning to submit a PR to transform 'columns' into a MAP, just like ProfileEvents map, allowing clickhouse to record more information without affecting current usage or bloating the query_log.